tests/tools/FFI/bluechi-tester

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+
+import argparse
+import sys
+
+from dasbus.typing import get_variant, Tuple
+from dasbus.connection import AddressedMessageBus, InterfaceProxy, ObjectProxy
+from dasbus.error import DBusError
+
+from gi.repository import GLib
+from typing import Any
+
+OBJECT_PATH_BLUECHI = "/org/eclipse/bluechi/internal"
+OBJECT_PATH_AGENT = OBJECT_PATH_BLUECHI + "/agent"
+
+INTERFACE_BLUECHI = "org.eclipse.bluechi"
+INTERFACE_AGENT = INTERFACE_BLUECHI + ".internal.Agent"
+
+SUPPORTED_SIGNALS = ["JobDone"]  # format: ["SignalName1, SignalName2, etc"]
+
+
+class BlueChiTester(object):
+
+    def __init__(self, node_name: str) -> None:
+        self.node_name: str = node_name
+        self.peer_bus: AddressedMessageBus = None
+        self.peer_bus_proxy: (InterfaceProxy | ObjectProxy | Any) = None
+
+    def run(self, timeout=5):
+        loop = GLib.MainLoop()
+
+        def _kill_loop():
+            loop.quit()
+            print("\nDone!")
+            sys.exit(0)
+
+        # Schedule a callback to exit the EventLoop after the timeout
+        GLib.timeout_add_seconds(timeout, _kill_loop)
+        loop.run()
+
+    def connect_to_controller(
+            self,
+            url: str,
+            signal_name: str,
+            num_signals: int):
+        self.peer_bus = AddressedMessageBus(f"{url}")
+        self.peer_bus_proxy = self.peer_bus.get_proxy(
+            INTERFACE_BLUECHI,
+            OBJECT_PATH_BLUECHI
+        )
+
+        self.peer_bus_proxy.Register(self.node_name)
+        parameters = get_variant(
+                Tuple[int, str],
+                (num_signals, "fake-state")
+        )
+
+        destination_bus_name = None
+
+        print(f"Sending {num_signals} {signal_name} "
+              f"signal(s) to controller... ")
+        for number in range(0, num_signals):
+            self.peer_bus.connection.emit_signal(
+                    destination_bus_name,
+                    OBJECT_PATH_AGENT,
+                    INTERFACE_AGENT,
+                    signal_name,
+                    parameters
+            )
+            print(f"  * Sending {signal_name} {number}...")
```

Comment:
NIT: We could think about moving this part to `run()` since its not really related to connecting to bluechi.

Reply:
sure